feature/experimental/isis/ate_tests/isis_change_lsp_lifetime_test/isis_change_lsp_lifetime_test.go

```diff
@@ -0,0 +1,317 @@
+func configureDUT(t *testing.T) {
+	t.Helper()
+	dc := gnmi.OC()
+	dut := ondatra.DUT(t, "dut")
+
+	i1 := dutPort1Attr.NewOCInterface(dut.Port(t, "port1").Name(), dut)
+	t.Log("Pushing interface config on DUT port1")
+	gnmi.Replace(t, dut, dc.Interface(i1.GetName()).Config(), i1)
+
+	i2 := dutPort2Attr.NewOCInterface(dut.Port(t, "port2").Name(), dut)
+	t.Log("Pushing interface config on DUT port2")
+	gnmi.Replace(t, dut, dc.Interface(i2.GetName()).Config(), i2)
+
+	if deviations.ExplicitPortSpeed(dut) {
+		fptest.SetPortSpeed(t, dut.Port(t, "port1"))
+		fptest.SetPortSpeed(t, dut.Port(t, "port2"))
+	}
+}
+
+// configureISIS configures isis on DUT.
+func configureISIS(t *testing.T, dut *ondatra.DUTDevice, intfName string, dutAreaAddress, dutSysID string) {
+	t.Helper()
+	d := &oc.Root{}
+	configPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, isisInstance)
+	netInstance := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	prot := netInstance.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, isisInstance)
+
+	if !deviations.ISISprotocolEnabledNotRequired(dut) {
+		prot.Enabled = ygot.Bool(true)
+	}
+	isis := prot.GetOrCreateIsis()
+	globalIsis := isis.GetOrCreateGlobal()
+
+	// Global configs
+	globalIsis.Net = []string{fmt.Sprintf("%v.%v.00", dutAreaAddress, dutSysID)}
+	globalIsis.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+	globalIsis.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV6, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+	globalIsis.LevelCapability = oc.Isis_LevelType_LEVEL_2
+
+	globalIsis.GetOrCreateTimers().LspLifetimeInterval = ygot.Uint16(lspLifetime)
+
+	// Interface configs
+	intf := isis.GetOrCreateInterface(intfName)
+	intf.Enabled = ygot.Bool(true)
+	intf.CircuitType = oc.Isis_CircuitType_POINT_TO_POINT
+	intf.InterfaceId = &intfName
+
+	t.Log("Pushing isis config on DUT")
+	gnmi.Replace(t, dut, configPath.Config(), prot)
+}
+
+// configureATE configures the interfaces and isis on ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+	t.Helper()
+	topo := ate.Topology().New()
+	port1 := ate.Port(t, "port1")
+	port2 := ate.Port(t, "port2")
+
+	i1Dut := topo.AddInterface(atePort1attr.Name).WithPort(port1)
+	i1Dut.IPv4().WithAddress(atePort1attr.IPv4CIDR()).WithDefaultGateway(dutPort1Attr.IPv4)
+	i1Dut.IPv6().WithAddress(atePort1attr.IPv6CIDR()).WithDefaultGateway(dutPort1Attr.IPv6)
+
+	i2Dut := topo.AddInterface(atePort2attr.Name).WithPort(port2)
+	i2Dut.IPv4().WithAddress(atePort2attr.IPv4CIDR()).WithDefaultGateway(dutPort2Attr.IPv4)
+	i2Dut.IPv6().WithAddress(atePort2attr.IPv6CIDR()).WithDefaultGateway(dutPort2Attr.IPv6)
+
+	isisDut := i1Dut.ISIS()
+	isisDut.
+		WithAreaID(ateAreaAddress).
+		WithTERouterID(atePort1attr.IPv4).
+		WithNetworkTypePointToPoint().
+		WithLevelL2()
```

Comment:
Please set link metric as well from the ATE using WithMetric()
By default the metric from ATE is 0.

Reply:
@arvbaska1, can you please address these requests in a different pull? We need to merge this one soon.

Reply:
Sure, we can address these comments with another PR.